In [30]:
import cv2
import pandas as pd
import numpy as np

def preprocesing(val=True):
    
    data = pd.read_csv('data/x_train_gr_smpl.csv')
    labels = pd.read_csv('data/y_train_smpl.csv')

    if(val):
        ## ---------------- Data preparation ---------------- ##
        X_train = []
        for i in range(data.shape[0]):
            img = np.uint8(data.iloc[i])
            edited = cv2.Canny(img, 10, 30)
            edited = cv2.GaussianBlur(edited, (5, 5), 0)
            X_train.append(edited.reshape((1,-1))[0])

        data = pd.DataFrame(X_train)
        ## -------------------------------------------------- ##

    return data, labels

In [2]:
from sklearn.cluster import KMeans
from sklearn.neighbors import NearestNeighbors
from sklearn.model_selection import train_test_split
from sklearn.metrics import homogeneity_score

In [3]:
n_clusters = [10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 50, 100]

data, labels = preprocessing()
labels = np.reshape(labels.values, newshape=(1, -1))

for n in n_clusters:
    estimator = KMeans(n_clusters=n, max_iter=50000, random_state=1, n_jobs=-1).fit(data)
    print(f'Number of clusters: {n} Homogeneity score: {homogeneity_score(labels[0], estimator.labels_)}')

Number of clusters: 10 Homogeneity score: 0.4950284618523185
Number of clusters: 11 Homogeneity score: 0.48341617311681173
Number of clusters: 12 Homogeneity score: 0.5167613058592444
Number of clusters: 13 Homogeneity score: 0.5204136214297174
Number of clusters: 14 Homogeneity score: 0.5530250196404715
Number of clusters: 15 Homogeneity score: 0.5568376753630506
Number of clusters: 16 Homogeneity score: 0.5833472105057467
Number of clusters: 17 Homogeneity score: 0.5763995636211059
Number of clusters: 18 Homogeneity score: 0.5815075350489741
Number of clusters: 19 Homogeneity score: 0.6004386672475748
Number of clusters: 20 Homogeneity score: 0.6146503165589815
Number of clusters: 50 Homogeneity score: 0.6949833069037257
Number of clusters: 100 Homogeneity score: 0.7471942791979834


In [43]:
from sklearn.mixture import GaussianMixture as EM
from sklearn.metrics import homogeneity_score
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler


data, labels = preprocesing()

# Transform data such that the distribution mean = 0 and std = 1
# scaler = StandardScaler()
# scaler.fit(data)
# scaled_data = scaler.transform(data)

# data = pd.DataFrame(scaled_data, columns = data.columns)
data['label'] = labels
data = data.sample(frac=1)

X = data.iloc[:, :-1]
y = data['label']

# X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.01)

y_pred = EM(n_components=10, max_iter=100, n_init=1).fit_predict(X)

KeyboardInterrupt: 

In [ ]:
print(homogeneity_score(y, y_pred))

=== Run information ===

Scheme:       weka.clusterers.EM -I 100 -N 10 -X 10 -max -1 -ll-cv 1.0E-6 -ll-iter 1.0E-6 -M 1.0E-6 -K 10 -num-slots 1 -S 100
Relation:     data_top_10
Instances:    12660
Attributes:   86
              1025
              1034
              2063
              2064
              1039
              1078
              1082
              1083
              2111
              2112
              1614
              1127
              1134
              2159
              2160
              1661
              1662
              1167
              1172
              1175
              1176
              1177
              1179
              1180
              1181
              1182
              1183
              1185
              1186
              1704
              1705
              1706
              694
              695
              696
              697
              1215
              1216
              1220
              1224
              1225
              1228
              1229
              1230
              2255
              1232
              1233
              1234
              1752
              1753
              742
              743
              744
              745
              1263
              1264
              1268
              1273
              1276
              1277
              1280
              1281
              1282
              792
              793
              1311
              1312
              1313
              1328
              1329
              1330
              1846
              1360
              1361
              1367
              1377
              888
              1415
              1416
              1417
              936
              1967
              1464
              985
              986
              label
Test mode:    split 66% train, remainder test


=== Clustering model (full training set) ===


EM
==

Number of clusters: 10
Number of iterations performed: 3


              Cluster
Attribute           0         1         2         3         4         5         6         7         8         9
               (0.11)     (0.2)    (0.08)    (0.04)    (0.07)    (0.13)    (0.07)    (0.06)    (0.14)    (0.11)
================================================================================================================
1025
  mean         54.3658    61.546   84.3844   36.2722   87.1532   56.5627    39.881   52.4302    44.481   75.8405
  std. dev.     35.139   34.6274    27.169   36.8463   27.4307   32.5913   41.2492   38.9559   35.7683   31.6741

1034
  mean          1.1458   51.3559   89.5964    0.8482  102.7246    0.3927   34.3119    42.248    15.663   82.1378
  std. dev.      6.718   38.9536   31.8021    6.2328   29.7257    3.7117   36.6951   41.0954   25.2357   35.2341

2063
  mean         75.8818    66.259     1.648    0.1696   70.5872   63.4836   58.6137    3.9913   56.9481   77.6871
  std. dev.     34.604   40.8459   10.5902    3.0277   38.0553   42.6564   43.5944   17.6744   43.2074   32.2809

2064
  mean         74.9367   65.1048    1.7452     0.254   69.6201   61.8222    55.599    3.6909   55.8412   74.3756
  std. dev.    33.5815   40.4846   11.0592    2.0296   37.8187   42.2139   41.8701   16.2234   42.4555   31.7192

1039
  mean         47.3232   64.5616   80.9789   31.0962   103.573    44.249   40.2446   37.6093   62.7604   74.6794
  std. dev.     36.472   36.5325   36.2837   35.7243   27.7127   33.3228   37.1502   40.1278   41.7304    38.868

1078
  mean          3.8293   57.2472    72.043    1.4594   86.0349    0.2157   38.2117   43.7979   32.9868   64.6047
  std. dev.    14.7048   39.2015   40.9057    9.2804     33.85    1.8582   36.9399   40.1088   32.9007   41.2652

1082
  mean          0.2976   55.7815   92.8734    0.3076  107.7692    0.6442   27.8885   43.6701   35.9023   83.5185
  std. dev.     3.0159   38.3729   32.8884    4.2216   25.5171    6.2722   34.6473   41.6963   34.8222   35.9075

1083
  mean          1.1756   54.6843   93.1498    0.2749  105.9954     1.143    30.409    45.523   23.9704   82.7046
  std. dev.     6.6024   37.2441   27.4329    2.0793   27.1199    9.4549   36.6537   41.4926   30.1758   34.1954

2111
  mean         75.4631   68.2978    0.1956    0.0334    72.557   63.4864   57.4622    1.3346   56.9857   78.9226
  std. dev.    33.7646   40.3854    1.7584    0.7303    38.143    42.933   43.5494    9.3583   43.0207   30.9433

2112
  mean         75.4478   65.8879    0.1187    0.1501   69.9813   61.2414   55.4041    1.0539   56.2212   75.0215
  std. dev.    34.1118   40.0263    1.3728    1.5426   38.0757   42.0257   42.6883    7.3797   42.4751   30.1309

1614
  mean         35.1594   32.4459   31.2479   32.4675    1.5543   72.9037   17.5644   29.8601   15.1676   31.8747
  std. dev.    36.0985   37.9673   34.6351   36.9951     8.804   37.8733   30.6869   36.3281   30.2452   36.1271

1127
  mean          1.1095   46.7451   76.5089    0.8155   89.4744    0.2017   39.2855    38.109   29.2698   68.1905
  std. dev.     7.4331   39.2269   38.2686    6.2068   26.4786    2.6487   38.4878   40.1194   30.8755   40.0193

1134
  mean         13.4969   65.0874    79.761   10.2824   94.0219   43.5426   42.6291   60.1547   73.9223   73.8835
  std. dev.    25.9398   34.8753   28.6637   22.2045   38.1297   33.2869   37.5661   40.5835   28.7364   33.5254

2159
  mean         77.2932   67.6393    1.6223     0.733   74.7413    63.574     56.92    1.3636    58.141   77.2921
  std. dev.    33.9951   40.4163   10.0204     6.866   37.5119   42.4857   43.9872    9.5203   42.7588   33.1605

2160
  mean         75.3209   66.0641    1.7692    0.9417   70.6461   62.5272   54.6748     1.373    57.333    73.587
  std. dev.    34.6341   39.6649   11.4552    7.9111   37.7691   42.0404   42.8768    9.3753   42.5314   32.1957

1661
  mean         35.2247   40.6522   28.3335    29.336    0.0089   71.5018   17.0333   21.6332   11.0808   25.3346
  std. dev.    35.3492    37.468    35.394   33.9904    0.3772   38.0931    30.132   34.0637   26.0652   35.8529

1662
  mean         42.0714   35.5555     36.73   32.4054    1.4914   76.2308   17.3899    22.149   10.6911   31.5721
  std. dev.    36.4449   39.9962   36.6899   37.0516    4.6517   37.7605   31.3525    33.073    25.949    36.607

1167
  mean         53.6057   61.3449   75.8731   41.5315   64.4208   64.3405   34.9743   59.8373    0.9412   70.7308
  std. dev.    35.7469   36.3406   23.4799   37.8477   31.8698   37.2019   36.3389   33.2115    5.0453   28.3755

1172
  mean          8.5065   62.0775   46.8722    3.1439    84.591   23.2701   51.7387   41.9318   47.3433   54.6872
  std. dev.    22.3698   36.2147   41.5249   11.4819    41.634   28.9847   37.1793   39.3846   35.3006   41.1393

1175
  mean          0.5632    41.155   83.4194    0.1479   92.2489     0.469   33.3353   34.5209   31.9528   75.7999
  std. dev.      5.017   38.4656   30.9801     1.531   22.1349    4.4689   38.0749    39.518      32.2   36.1233

1176
  mean           0.091   37.4473   89.5748    0.1386  101.0685    0.0661   25.6798   32.3518   27.7602   80.8231
  std. dev.     1.4541   36.7933   29.8147    2.9754   20.1492    1.0261   34.5504   39.6804   30.5776    34.894

1177
  mean          0.0035    46.223   94.8146    0.2079  106.8118         0   24.0873   35.3807   39.9295   86.8163
  std. dev.     0.2356   37.0439   29.8136    4.4631    22.157   45.7299   32.3671   39.7985   35.0325   34.5203

1179
  mean               0   61.3762    95.321    0.0347  104.5829    1.7534   43.1172   58.2026   58.1194   88.5185
  std. dev.    46.1442   37.3068   26.2611    0.7438   27.1725    8.0608    38.354   39.6864   37.4068   31.9037

1180
  mean               0   59.9916   87.1209     0.131  102.9022    8.8098   47.1504   65.1453   54.7952   85.5018
  std. dev.          0   36.3211   23.1145    1.4418   27.2405   18.4391   38.1228   38.4621   35.8646   27.9673

1181
  mean           1.084   64.6958   75.7234    1.6077   96.0028   28.3067   42.1525   64.3507   61.4112   76.2079
  std. dev.      4.021    34.686   26.4473     6.949   35.7861   29.8895   38.3803   38.6264   32.7144   30.9553

1182
  mean          7.1221   68.5065   79.1383    7.7071   92.6384   56.3051   35.6495   60.9238   73.4812   73.3895
  std. dev.    16.4727   34.1251   27.6107   18.4755   39.0865    32.833   37.4154   39.3243   30.1889   33.4144

1183
  mean         21.3026   64.6113   91.0124   20.1557   98.4781   71.6399   30.3811   59.7519   74.5674   81.5292
  std. dev.    30.9124   33.4823   26.0614   31.5427    31.204   28.1789   37.0725   40.0684   29.5172   31.5249

1185
  mean         56.2953   71.2552   97.8219   44.6956   93.2366    62.265   16.1931   68.0491   66.2509   90.4536
  std. dev.    34.4938   33.3931   24.2837   36.2001   28.8088   34.9739   27.6682   37.4083   31.2153   28.2052

1186
  mean         62.3254   73.5444   95.1543    47.513   77.7343    70.459   15.4172   70.6338   56.9048   88.3004
  std. dev.    32.0044   33.8612   21.5617   36.6219   36.2332   38.0603   27.1446   34.0925   34.9653   29.3667

1704
  mean         44.8476   58.6817    5.1899   24.3579    0.1809   79.4815     9.302    2.3961     6.066    6.2394
  std. dev.    46.9213   47.9915   17.8965   38.3606     2.833   40.6326   24.1843   10.4386    21.441   19.2254

1705
  mean         43.5133   60.1467    6.2651   23.9059    0.0269   80.4695    9.2584    1.1161    4.9917    6.4199
  std. dev.     46.878   49.5322   20.1382   38.1181    0.6554   41.8809   24.7148    5.5646   19.9677   20.4036

1706
  mean         42.5847    60.372    11.488   26.6533    0.0385   78.4241   11.8876    3.2375    4.5974    9.7673
  std. dev.    43.0932   49.5485   26.7363   37.8873    0.7844   38.3143   27.6658   12.5584    18.753   24.8844

694
  mean         50.8204   61.9711   17.8944   28.8682     0.017    1.1622   14.0429   10.4333    1.5969    16.521
  std. dev.    42.1616    42.983   31.3127   38.2922    0.5208    8.8121     29.97   24.9295    9.3171   30.2944

695
  mean         49.9552   63.1161   10.5381   25.0387         0    0.7406   10.3948    5.3146     0.402    9.1161
  std. dev.     45.779   44.4283    24.875   38.5346   38.7222    6.4735   25.3018    16.659    3.4068   22.7827

696
  mean         46.3216   61.3919    6.3871   23.9608         0    0.8354    9.4201    3.9847    0.2871    5.0206
  std. dev.    45.4085   45.2927    19.572   37.9242         0    7.2935   24.8187   14.6977    3.4124   15.7349

697
  mean         43.6293   57.1299    6.5124   24.9034    0.4633    1.3907   12.4481    6.8488    0.9023    5.9109
  std. dev.    42.4638    43.937   19.4755   37.7847    2.6829    9.4413   28.6686   20.3296    7.2602    18.183

1215
  mean         52.5873   62.5736   79.4043   39.3592   75.2163   68.6054   36.4126   60.6478    0.4206   72.6734
  std. dev.    35.6265   35.8059   22.6933   37.1529   27.9108   38.3353   37.2174   33.8829      2.56   28.2418

1216
  mean         41.1611   61.9877   80.6333   34.8708   74.2733   61.5499   42.4942   57.7463    0.1803   74.0055
  std. dev.    36.6931   34.9328   23.8712   36.2945   27.9404   35.9303   38.5572   36.8836    1.6887   29.6946

1220
  mean          9.0212   60.6985   53.2596    3.4877   96.2263   37.5037   51.3638    36.778   26.5812   55.3407
  std. dev.    23.2842    35.779   44.3039   12.7909   32.7603   33.0947   39.2822   39.4232   31.8931   43.7864

1224
  mean          0.2537   35.1722   91.3505     0.483  103.1149    0.1452   25.8169   29.7076   32.2856   84.4165
  std. dev.     2.9315   36.2138   34.9667    5.2836    20.902    1.5175   32.9796   39.5438    32.115    37.615

1225
  mean           0.255   43.0499   92.1296    0.3868  108.2369    0.0101   30.7208   33.2011    26.748   86.9672
  std. dev.     3.8907   37.2925   36.7417    5.1976   22.3236    0.4025   35.3179   39.2395   30.5758   37.1969

1228
  mean          0.1018   58.4395   89.3426    0.4984  102.9849   17.2359   41.5419   67.3528   58.3005   86.0092
  std. dev.     1.2722   35.5116     21.59    2.7796    26.853   24.8632   38.6235    36.677   39.4828   27.3427

1229
  mean          1.7855   62.0778   76.9697    3.4799   96.4884   43.1042   35.4675   63.8818   62.7931   75.9772
  std. dev.     5.8195   34.7481   26.3242   11.5227   34.9515    33.058   38.1096   37.5758   38.8189   30.9735

1230
  mean           9.578   68.0392   78.1021    11.863   93.5008   66.7873   30.8296   59.8867   69.3903   72.4071
  std. dev.    19.8445   34.3732   28.0138   24.4114   38.5009   31.0624    38.656   38.9385   37.2236   34.0478

2255
  mean         73.6768   66.7378    8.5119    8.9475   71.9453   64.5184   54.9342    7.6378   56.6277   72.3839
  std. dev.    36.3284   40.6368    25.979   27.3232    39.325   42.3828   43.8238   23.8797   43.2284   38.0079

1232
  mean         45.1497   62.9513   97.0811    37.052    103.99   63.0188   17.2368   63.3839   67.9948   89.5296
  std. dev.     36.575   33.1713   26.3758   37.6708   24.9198   33.5653   30.0182   39.1025   33.0094   29.0878

1233
  mean         60.6001   69.0311    98.176   44.8147   98.7783    65.401   10.3551   69.1378   67.0566    92.232
  std. dev.    32.3317   32.7106   24.1649   36.2365   27.4813   37.8116   21.1638   36.5304   31.8901   26.9842

1234
  mean         62.6135   73.8952   95.3171   45.4688   87.6784   75.3559    11.925    72.204   58.7973   90.0865
  std. dev.    32.0769   33.6465    21.336   37.7334   32.0246   38.0513   24.2563   33.9668   34.7703   27.6405

1752
  mean         30.0188   40.9032    7.2218   18.3924    1.1221   64.9372    8.7159    1.6621    2.0807     7.967
  std. dev.    42.1251   44.1468   20.6334   32.7504    8.6542   46.9863    23.583    7.9353   12.1285   22.1512

1753
  mean         27.4553   43.5718    7.8344   17.5913    0.8405   65.6626    9.3363    1.2965    1.7451    7.4432
  std. dev.     40.734   46.7434   21.6967   32.4646    7.7754   48.8213   24.2569    6.2937   11.1997   20.9602

742
  mean         54.6702   60.4714   15.9673   27.8261     0.128    0.1728    14.838    12.106    2.3239   15.6487
  std. dev.    37.8342   39.9598    29.684   36.8602    1.4256    2.1543   31.2656   25.6204   11.3206   29.6956

743
  mean         48.8358   61.1851    9.1355    24.255         0    0.0099   11.1835    6.4805    0.7747    7.2033
  std. dev.    41.4563   41.1831   22.8249   37.5997   36.7368    0.3985    26.615    19.012    6.0325   18.9715

744
  mean         44.8691   61.2718    5.3124   23.3207    0.0867    0.0098   10.8521    4.7828    0.5122    2.9231
  std. dev.    41.1793   41.6506   17.9632   36.7157    1.1746    0.3958   25.4941   15.8337    5.0369   10.9277

745
  mean         48.3712   59.3055      5.45   26.6654    0.5997    0.2478   13.9813    7.8535    0.9343    3.7343
  std. dev.    40.2602   41.1761   18.1049   38.0778    5.0875    2.4051   29.0683   20.0141    7.9366   14.2284

1263
  mean         56.3965   62.8578   85.9115    39.004   87.7817   73.3384   38.6021   59.8899    0.3855   79.9147
  std. dev.    34.6737   35.2788   22.4832    36.347   26.5467    38.161   37.7967   35.1826    2.4534   28.2742

1264
  mean         48.4862   64.0072    87.732   35.1339   90.7766   66.0797   44.8517    56.414         0   80.9749
  std. dev.    36.6719   34.9202   23.8073   36.9382   23.2671   37.8479   38.9111   39.3442         0   29.8893

1268
  mean         11.1684   57.1582   73.2654    6.2845  101.8516   53.6571   42.9251   39.9158    9.4129   70.2034
  std. dev.    26.1078   36.6316   39.2033   19.0221   25.9385   33.9871   40.7199   41.2379   21.0601    41.033

1273
  mean          0.9678   39.8378   84.4278    0.5968  109.4297    0.2044   37.4634   29.2821   36.2303   81.5469
  std. dev.      8.443    36.629   44.9982    6.3091   22.7597    1.7968   37.0645    37.146   33.0285   44.1315

1276
  mean          1.4646   55.9524   88.9537    1.6566   101.887   28.0547   34.4577   66.4209   52.0274   84.9848
  std. dev.     6.8015   36.0717   22.4819    7.8019   27.4284   30.1838   38.0983   37.7085   43.4008   28.9279

1277
  mean          6.6003   56.7928   76.8924    6.8357   95.4219    55.746   28.0245    59.766   58.3742   75.0619
  std. dev.    16.8705   36.7022   26.5136   17.1633   35.4356   33.0132   35.9847   38.1245    46.544    32.712

1280
  mean         54.9993   68.6148   95.8107   42.3582  105.0926   63.0879   12.1112   60.3013   64.9167    90.342
  std. dev.    34.6753   31.7046    27.529   37.2369   26.5779   34.7788   25.6988   39.2883   36.2613   29.3532

1281
  mean         61.4971   69.9286   98.0027   43.0598  105.1133   70.7441     6.686   65.3438   67.5982   93.3451
  std. dev.    32.2916   32.1652   25.2679   36.4132   26.9859   37.8819   16.4621   37.2939   32.7279   27.7264

1282
  mean         53.9004   72.0212    96.833   39.1164  100.2317   77.9083    9.7266   69.5063   60.1699   91.1038
  std. dev.    33.5323   34.5199   21.5264   36.2378   25.6251   36.9056    21.122   35.1346   33.9404   27.9131

792
  mean         35.9236   54.0299    4.6472   19.5789    7.5558    0.1055   17.7509   12.0021    0.6938    5.1155
  std. dev.    38.6802   39.3955   13.9478   32.7291    20.306     2.198   33.1996   25.9866    5.3477   15.6561

793
  mean         42.5973   56.2325    6.4087   23.3687   13.9921    0.2175   19.9263   15.5797    0.9704    6.4278
  std. dev.    38.6922   39.7726   17.6841   34.4155   28.4397    2.3109   34.0761   28.6779    7.8731   18.0372

1311
  mean          55.977   64.5993   90.1679   36.6689   96.8884   72.4678   37.2106   57.0618    0.4452    83.596
  std. dev.    35.0719   36.4813   22.3078   36.2201   26.2564   38.4368   37.1419   35.4356    2.6316   29.4193

1312
  mean         56.2331   67.8912   91.7837   36.5321   97.2747     70.48   41.1001   53.2884         0   84.1919
  std. dev.    35.7388   36.2152   25.8785   36.7551   23.5536   38.5031   37.7323   39.1704   42.8789   30.9057

1313
  mean         46.7464   62.4956   89.9299   33.9782   90.4902   63.5984   44.7938   46.9838    0.0923   80.3351
  std. dev.    36.6114   36.8368   29.1572   37.6644    24.546    36.842   39.3589    40.801    1.2117   33.3467

1328
  mean         59.6567   72.1545   94.5806    41.368  103.7565   65.1416    8.4173   59.1432   54.8583    89.037
  std. dev.    33.2892   31.7325   28.8762    36.778   27.6244   36.9666     21.32   39.5806   38.3323   31.4317

1329
  mean         54.0314   68.6179   96.6966   37.7425  104.4108    74.294    5.0193   64.5364   64.0564   92.0431
  std. dev.    33.6996   32.7243   26.6243   35.9275   27.7842   37.3601   13.0249   37.3944   33.3883   29.7285

1330
  mean         42.7961   66.6761   95.8227   34.7863  102.2237   76.6971    9.9338   67.9251   60.2037   90.2304
  std. dev.    35.8234   35.2255   22.5304   37.4704   23.9416   37.7996   21.8336   36.2177   33.5637   29.0771

1846
  mean         49.0553   32.7893   13.6325    35.911    9.9462    71.326   16.7947    7.3393    7.4365   19.2209
  std. dev.    37.6509   41.4593   28.4138   38.6445   26.0735   39.3346   31.8292   21.6198   22.3344   33.0438

1360
  mean         54.4816    69.318   91.4009   33.9592   97.6459   72.1195   40.1856   51.4855    0.0452   82.9497
  std. dev.    35.5517   38.9213   28.1402   36.4164   26.0687   38.3008   37.7057   38.4259    0.8492    33.543

1361
  mean         55.8857   68.6592   87.9589   35.2665   90.8578   65.9537   40.1616     44.27    0.1013   77.6267
  std. dev.    35.8966   38.0331    31.801   36.8155    24.987   38.0564   38.4969   40.5441    1.2691   35.8109

1367
  mean          4.8781   32.1238   85.9625     1.251   92.6584   10.7563    37.595   35.3586   24.5804   81.6686
  std. dev.     16.694   37.2224   39.2192    8.4166   25.9418   21.4877   37.1583    40.461   32.8064   40.3999

1377
  mean         41.9425   67.9589   92.9273   30.5485  103.8168   76.6407    8.1604   60.7305   58.8936   90.4099
  std. dev.    34.5598    33.373   28.7646   34.5368   28.1963     37.21    19.899   37.3329   33.5068   31.2504

888
  mean         13.9059   35.7564   36.5821    7.8374   70.9789    0.2191   21.9786   34.9586    5.9986   35.9867
  std. dev.     27.414   37.1471   33.4963   20.5035   41.0006    3.0197   32.6291   37.5991   17.9365   36.5952

1415
  mean         10.6929   33.2272   88.8153    3.0718   82.7862    20.614   41.8837    45.347   18.0001   82.0147
  std. dev.    25.4398   37.1342   31.5746   11.1845   27.8709   27.8478   37.6005    39.739   30.5971   35.6828

1416
  mean          5.2304   27.9132   88.7023    0.4927   87.6673    7.2903   41.2538   44.2863   23.7223   83.3361
  std. dev.    17.3045    35.168   34.2428    2.7641    23.207    17.728   37.4932   39.6972   31.4454   37.0675

1417
  mean          2.8613   32.9987    86.939     0.966     90.84    6.2837    36.447   43.5904   36.7486   81.1156
  std. dev.    12.6153   36.4808   34.6449     4.872   23.4213   16.1045   36.5763   40.6229   34.8489   38.5284

936
  mean          7.4369   33.7939   60.2561     3.351   89.4175    0.0489   26.5713   41.9065   14.6936   54.6219
  std. dev.    21.0517   38.1042   33.8486   13.6604   32.0137    0.8835   34.1797    38.903   25.3735   37.1752

1967
  mean         72.1377   64.3323    8.6434    7.1018   69.2604    64.411   57.0487    7.5757   56.0997   70.5723
  std. dev.    36.9453   40.7287   25.5435   23.7253   40.1045   43.1939   43.5916   24.0263    43.249   38.4601

1464
  mean          9.4673   29.5538   72.3416     3.656   65.2458   15.8952   37.6419   48.4929   20.6485   71.3825
  std. dev.    22.4652    34.713    33.239   12.7678   28.7671   25.4486   37.0728   38.7024   31.1217   35.7939

985
  mean          2.5457   42.2888   85.2679    0.8578  100.3724    0.0099   41.0951   44.6245   14.4864   75.3126
  std. dev.     11.048   40.0414   29.3707     5.739   25.4004    0.3985   37.6935   40.2038    25.098   34.4873

986
  mean          5.2339   46.9851   87.2566    2.3272    94.761    0.1442   42.0064   43.8117    6.7895    77.375
  std. dev.     16.219   38.8642   28.2653     10.53   31.8951     1.512   37.6848   39.7407   17.6772   34.5514

label
  eight         7.2538    296.78         1   22.9023         1    1.0077   62.1808    17.902 1616.1251   53.8483
  one          10.6545  173.3046    547.97   25.2485   150.982    3.5149   90.1417  312.0689    6.8758  549.2392
  three              1 1274.9405    3.2565         1         1         1    9.0066   28.8659    2.0015     7.929
  six                1   42.3584    5.0236         1  624.2017         1    4.7257    2.4064         1  107.2841
  five         117.925  295.0449         1   51.2984         1 1578.1701   67.3345    7.3373   46.9602    3.9295
  two                1   19.3742   35.8295         1    1.9995         1    17.718  272.3418   21.7766   57.9603
  four       1246.6631  396.6689         1  328.8674         1   28.1183   75.4443   19.3758    7.1079    5.7543
  zero          3.8451   86.2813  437.1653   40.9267   99.1733    5.0776   41.6381  118.9484    9.8398  577.1044
  seven         8.5854         1         1    9.0042         1         1  210.1997    9.5132    7.6974         1
  nine               1    3.2157         1         2         1         1  261.0279    3.0265   24.8695   11.8604
  [total]    1398.9269 2588.9685 1034.2448  483.2476  882.3566 1620.8886  839.4174  791.7864 1744.2537 1375.9094


Time taken to build model (full training data) : 20.05 seconds

=== Model and evaluation on test split ===

EM
==

Number of clusters: 10
Number of iterations performed: 4


              Cluster
Attribute           0         1         2         3         4         5         6         7         8         9
               (0.12)    (0.11)    (0.06)    (0.06)    (0.07)    (0.08)    (0.19)    (0.13)     (0.1)    (0.08)
================================================================================================================
1025
  mean         53.7222    47.045   49.0206   50.4827   29.7904   78.9373   63.6165    55.308    85.363   81.8882
  std. dev.    34.8483   35.0523   40.1749   38.8376   38.4099   31.1564   33.6062   32.9014   27.3428   30.2361

1034
  mean          0.4692   16.4364   37.8566   41.4024   17.1485   87.8658   49.8964    0.0912  100.2096   81.8101
  std. dev.     2.6995   24.4998   38.5957   39.5322   29.5804   31.2092    38.921    1.2046   30.5057    34.343

2063
  mean         61.5188   59.4952   65.9111   35.6992   28.6024   74.1852   66.3555   61.3165   64.7717    0.2183
  std. dev.      42.99   42.3177   40.5026   43.1059   41.9438    35.351   41.1297   43.1924    40.959    2.8468

2064
  mean         60.1942   58.2579   64.8374   33.0072   27.1389   70.7354   64.8611   59.6028    63.559     0.364
  std. dev.    41.9238   41.6112   39.6548   40.6434   39.4928   34.4926   40.8523   42.5983   39.9613    4.4348

1039
  mean         46.6243     67.42   43.3445    46.538   27.0411   79.7286   66.1358   43.5791   94.8097   73.5494
  std. dev.    36.3731   39.8321   36.0536   41.4538   36.3035   38.2996    35.142   33.3616   31.2394   40.9242

1078
  mean          2.2531   33.4163   34.9386   47.1025   26.7857   72.9902   56.9395    0.0874   75.5078   69.4585
  std. dev.    11.1622   31.7046   36.2848   40.2028   35.3798   38.2947   39.3408    1.1796   42.1857   39.3092

1082
  mean               0   40.3863   31.6036   47.2123   15.7906   88.6957   54.4325    0.4125  103.7216   87.7056
  std. dev.    46.3377   34.2644    36.758   40.2943   29.1997   32.2293   38.4909    4.9767   28.2453   34.7806

1083
  mean          0.4323   25.9924   34.5447   45.9131   17.2487   88.4191   53.8949    0.6244  101.3509    87.983
  std. dev.     2.5941   29.8884   38.4073   40.1498   29.9873   30.3735   37.1739    7.2174   28.2614   31.0078

2111
  mean          61.391   58.6116   64.8455   35.2326   28.5947   73.9578   67.5372   61.0989   65.6611         0
  std. dev.    42.7204   42.3519   40.8385   43.2891    42.129   35.4306   40.9922   43.0413   41.4137    0.0002

2112
  mean         59.7824   57.7186   64.5055   32.4206    26.379   69.8143   65.7807   59.6203   63.1921         0
  std. dev.    42.1405   41.5041   41.0854   41.1556   38.8735    34.109   41.1398   42.6258   40.7782   43.7768

1614
  mean         35.5961   14.5999     28.99    35.386    8.9127   34.5624   33.3134   71.8043    7.4527   35.0623
  std. dev.     36.161   30.3259   36.6339    36.456     23.12   36.8814   38.4653   38.0636   21.1691   35.0355

1127
  mean          0.4678    30.213   35.7924   43.4994   21.1538    75.313   45.9887    0.0566   80.6488    72.869
  std. dev.     4.4158   30.6215   38.5827   39.9166   32.7073   37.5116   39.3037    0.9497   35.5058   39.2961

1134
  mean         11.9717   78.3763   43.7817   62.7435   39.4191   82.2054   64.1735   42.1733   81.3996   81.9054
  std. dev.    23.8606    23.967   37.6512   38.7384     39.43   31.1907    34.731   33.4504   39.5585    29.242

2159
  mean         62.1925   60.8504    61.828   35.8076   29.9622    74.675   66.3845   62.0314   67.3619         0
  std. dev.    43.0041   42.0995   41.6218   43.7555   42.7654   35.1386   41.2143   43.4288    41.809    0.0001

2160
  mean         59.5665   59.9988   61.6614   33.2101   28.1272   71.0682    65.491    60.767   63.1228    0.0006
  std. dev.    42.5462   42.1341   41.4078   41.6841    40.279   34.2852   40.8044   42.3938   40.4405    0.0941

1661
  mean          35.371   10.5256   28.4216   28.6775    5.3434   33.4057   43.0591   71.7043    0.7228   29.5937
  std. dev.      34.75   25.3987   35.6761   37.4677   17.5287   38.1826   37.8025   38.0849    5.8426   35.3672

1662
  mean         42.6334   10.3315    28.674   29.8646    6.4202   41.6555   37.8562   75.6512    2.2456   38.1611
  std. dev.    36.0098   25.1286   37.1646   36.8068   19.2728   38.0894   40.8399    38.572     5.575   36.4667

1167
  mean         53.4402    0.3943   41.1337   49.0562    27.138    73.762   63.4059    62.919   68.1423   75.1121
  std. dev.    35.6746    2.4806   35.8838   37.8482   35.8944   26.1813   35.7272   37.5949   29.5975   24.4547

1172
  mean          5.6554   49.8579   51.8573   44.5137   30.7851   59.2169   61.4563   22.0493   69.8163   50.2604
  std. dev.    17.7257   34.1977   38.8714   39.4814   36.1356   41.5898   36.0475   28.4766   45.1231   41.9734

1175
  mean          0.1493   34.7938   36.3047   43.1463   12.9227   82.1753   40.7016    0.1277   87.3678   78.3676
  std. dev.     1.5381   32.2979   39.6078   40.5733   25.1074   33.2111   38.5403    1.4235   25.6684   35.3318

1176
  mean               0   29.6087   30.1773   41.9691    8.4038   88.7137   36.0996         0   94.8542   82.2552
  std. dev.    43.8689   30.1065   36.7113   39.7743   20.4774   32.8383   36.2693     0.001     23.83   36.8067

1177
  mean               0   43.2685    28.074   44.0753    10.759   94.1082   44.4527         0  103.0426   85.7657
  std. dev.          0   33.6735   34.3235   39.0324   24.1999   32.3451   36.8241    0.0042   23.3249   36.9287

1179
  mean               0    65.149   43.9232   56.7788   28.9346   93.9242   59.4878     1.302  100.7384   89.4849
  std. dev.          0   34.9148    38.661   38.1246   37.4807   28.6974    37.907    6.0118   27.7151   31.1635

1180
  mean               0   59.3264   46.9415   60.4694   36.5383   90.8329   58.2691     8.004   95.3649   86.6286
  std. dev.      43.73   34.4892   39.2854   38.5888   39.9822   24.5328   36.5958   16.9845    28.332   25.3475

1181
  mean          0.7509   64.2594   41.3413    63.171   39.1203   83.8195   63.6011   27.0767   83.5547   80.5418
  std. dev.     3.3839   31.2099   39.0391   39.6299    38.996   27.4964   34.6915   29.7286   38.4582   25.4896

1182
  mean           6.016   77.6164   36.7482   64.2221   38.6004   83.0584   67.7139   54.5869   80.3909   82.6336
  std. dev.    14.2911   26.4997   37.6644   38.4171   39.1155   30.5095   34.1115   33.3563   39.9879   26.8881

1183
  mean         19.8453   79.9619   35.4833   63.1462   36.8928   89.9472   63.9497   70.0804   90.9454   89.7045
  std. dev.    29.5243   24.0513   38.7089   38.4554   38.9128   28.3698   33.1351   29.2311   31.8852   29.1776

1185
  mean         56.1056   71.3487    24.063   66.5177   33.2599   93.2358   71.9453   61.5704    93.669   95.3492
  std. dev.    33.8102   26.5512   31.9021   35.3194   39.0218   26.3554    32.848   34.6916   27.9542    25.903

1186
  mean         63.6564   61.3961   23.0198   65.3889   31.5572   89.6274   74.3887   69.9561   83.4441   92.9615
  std. dev.    31.4479   33.3176   31.6106   33.5208   39.1177   28.1695   33.4133   38.1829   33.3039   23.3193

1704
  mean          42.905    6.3956   16.9854    3.7591    1.0563    10.791   64.8011   78.7781    0.0745    5.3105
  std. dev.    46.5665   22.2589    30.759   13.5432    7.2537   25.5299   47.0169   40.6698    1.0895   17.7226

1705
  mean         41.5759    4.7535   18.0411    2.4742    0.2742   11.9474   66.1574   79.3682         0    5.9469
  std. dev.    46.2112   19.1231   32.5188    8.5809    2.0765    28.098   48.3097   42.3177   45.2779   18.9135

1706
  mean         41.0425    4.0547   23.1937    6.9222    0.7399   16.4907   65.2399   77.1592         0   11.1041
  std. dev.    42.4595    16.761   36.2986   18.4728    5.2062   32.0535   48.6884   38.3645   44.2121   25.8211

694
  mean         50.0128    1.0643   21.5906    22.983    0.3775   23.0308   63.6951    0.6327         0   18.4553
  std. dev.    42.3393     7.246    33.764   34.9395    2.4285   34.3162   42.5924    5.2886   38.6737   31.4934

695
  mean         47.9244    0.1907   16.7851    14.451         0   13.7715   65.3491     0.159         0   10.2695
  std. dev.    45.4932    1.7362   30.3858   26.7221   38.5623   27.7526    44.346    1.6791   38.5623   23.9726

696
  mean         44.0107    0.0391   16.3653   10.5202         0    8.0016   64.1724    0.3797         0    6.3021
  std. dev.    44.9484    0.7902   30.4922   23.3917   37.6553    20.142   44.9829    4.2154   37.6553   19.3356

697
  mean         41.6846    0.4559   21.3108   13.0524    0.0774    9.2646   59.7997    1.0603    0.3299    6.9791
  std. dev.    41.9205    3.8208   34.4583   27.3614    1.1102   22.3196   43.5624    8.4068    2.2736   19.7012

1215
  mean         51.6608    0.2164   42.7442   50.5685   26.3317   76.3465   64.4406   66.8762   75.8164   77.9664
  std. dev.    34.9188    1.8483    36.971   37.2938    35.626   25.0663   35.3325   39.4425   27.2652   24.5345

1216
  mean         40.3556         0   49.0376   49.4157   26.2715   78.8136   62.6187   60.7011   74.2465   79.1927
  std. dev.      36.18    0.0008   37.1779   38.7942   36.5404    25.055   34.4469   36.1819   27.8797   25.3591

1220
  mean          5.9698   25.9885   50.4275   42.2079   27.9313    58.962   60.0067   35.9278   81.1965   52.3658
  std. dev.    18.6922    30.844   39.8208   40.6348   36.0438   44.6935   35.3066   32.7716   42.7959   43.2315

1224
  mean          0.0796    35.108   30.9195   41.5776    8.3736    91.832   33.4867     0.057   99.7099   82.0302
  std. dev.     1.1259   31.7452   35.3466   40.3614   19.4187   35.6924   35.5545    0.9536   23.9997   40.7227

1225
  mean               0   28.7197   34.3483   43.7559   10.6118   93.5948   41.1272         0  104.4236   82.9016
  std. dev.    46.2594   30.6902   36.9121   38.9889   23.1567   35.6484   36.6892   46.2594   24.5745   42.1034

1228
  mean               0   63.2319   36.1088     64.02   39.4383   90.3556   56.8253   15.9734   95.4764   89.0273
  std. dev.    43.4538    38.068   38.4479   36.5478   40.9367   25.0052   35.3867   23.6483   28.2008   22.3874

1229
  mean          1.6424   66.9047   31.3894   66.4571   38.6757   83.1104   61.1456   41.5243   83.2303   80.9062
  std. dev.     4.8561   37.2564   36.8734   38.4992   38.8833   28.6928   34.0645   33.2438   38.1438   25.0739

1230
  mean           9.144   74.4166   31.8588   64.9257   35.9474   82.2445   67.6385   65.4596   79.7266   81.5069
  std. dev.    19.4433   34.5627   38.4942   38.7192   38.5803    31.244   34.0342    31.907    40.734   27.3691

2255
  mean          61.718   59.7029   60.4405   38.5981    31.313   68.9679   65.2731   63.0864   66.5809    6.0392
  std. dev.    43.2396     43.09   42.0644   44.1158   42.7442   41.3256   40.6301   43.3217   41.9618   22.0169

1232
  mean         43.7883    75.266   26.8765   63.7046   32.4456   95.3369     63.15   63.4223   99.4846   94.6437
  std. dev.    35.9952   27.4011    35.632   35.9342   39.9314   27.4748   32.7924   33.6331   25.1729   30.0706

1233
  mean         60.6317   73.8242   18.8378   69.4793   30.7127    95.666   69.9463   65.2536   97.9751   95.7714
  std. dev.    31.8823   26.1232   27.6432   33.0516   38.9112   26.3704    31.855   37.7883   26.6377   26.2708

1234
  mean         64.0782    63.767   19.5969   68.5101   30.1339   92.4113   75.0818   74.5546   90.1044   93.4188
  std. dev.    31.3777    31.907   29.1789   32.7494   39.2021   26.3603   33.0169   38.2819   29.3821   23.0065

1752
  mean         29.2419    2.2038   18.6814    2.6893    0.1996   12.7114   44.2211   63.8524    0.1361    7.4701
  std. dev.    41.8291   12.4767   32.2351   10.2289    1.7759   27.9349   44.8603   46.8901    1.4695   20.4883

1753
  mean         26.6111    1.6511   19.0839    2.7492    0.2349   12.7663   47.0883   64.4483         0    7.8214
  std. dev.    40.2659   11.3668   32.8849   10.1649    1.9244   27.5022   47.2904   48.8238    0.0253   21.4558

742
  mean         54.4472     1.559   23.1884   24.7099    0.0458   21.3195   62.5579    0.0457         0   17.0852
  std. dev.    37.5939    7.7597   35.6704   34.2658    0.8544   33.5952   39.5872    0.8541         0   30.2059

743
  mean         48.0224     0.216   19.4346   14.5489         0   11.0366   64.3136         0         0    9.9762
  std. dev.    41.5702    1.9315   32.5672   27.1284   37.0074   23.4918   40.9291   37.0074   37.0074   23.8236

744
  mean         44.2253    0.1397   19.9332    9.6008         0    5.2459   64.5114         0         0    6.2558
  std. dev.    41.2279    2.3457   32.1845   21.0804   36.3762   15.1832   41.4957   36.3762   36.3762   20.1565

745
  mean         47.9379    0.5132   24.6074   12.2911    0.3731    5.8915   61.5826    0.1066    0.2982    6.9765
  std. dev.    40.2768    5.4328   35.5863   25.1104    2.4147   18.4191   40.7885    1.3017    2.1639   20.6083

1263
  mean         56.4227    0.1974   43.4981   50.8013   25.1573   84.2857   64.3051    70.815    85.968   84.3169
  std. dev.    34.1406    1.7662   37.5396    38.305   34.4468   24.6452    34.729    39.508   27.0648   24.1521

1264
  mean         47.4977    0.0173   51.2448   49.0827   24.8014   85.9554   65.0452    63.822   89.1351   85.4485
  std. dev.    36.5117    0.5265   37.5583   40.1224   35.9617   23.9815   34.0316   38.8367   25.3889   26.4167

1268
  mean          8.4262    7.6318   43.6081   42.0629    24.166    74.742   57.4385   52.2616   93.8051   68.7404
  std. dev.     22.597   18.2847   41.5566   42.7519   35.6692   41.0368   35.6891   34.2876    30.451   42.3421

1273
  mean          0.2339   40.1979   39.0597   38.8407   14.9769   84.4454   37.6619    0.1342  105.2821   71.9444
  std. dev.     3.6047   32.6063   37.7447   38.3282   27.5975   43.2184   35.6441     1.459   28.4357    48.384

1276
  mean          1.2525   54.8179   27.9104   61.3973   39.3711   90.4906     55.56   27.2702   95.0448    87.962
  std. dev.     5.6526   42.4405   36.1524   38.5975   40.5038   24.6528    35.761   29.8709   28.7236   23.7848

1277
  mean          6.3317   62.2958   25.1404   60.7128   33.9547   82.5793   56.5037   54.6381   82.4121   79.6081
  std. dev.    16.4544   46.1313    33.936   41.1709   38.0305   29.3967   36.4346    33.726   38.2468   25.9493

1280
  mean         53.4366   70.2751   22.7891   64.1177   29.9281   96.0619   69.0513   62.7229   100.689   92.4327
  std. dev.     34.589   33.1648   34.0187    35.396   38.8749   27.1493   31.1567   34.4035   26.3035   30.9706

1281
  mean         61.9099   72.8671   14.3826   68.3079   30.2982   96.6142   70.7535   69.1968  102.8461   95.3862
  std. dev.    31.8335   27.7891   23.7721   33.6844   38.5242   26.9508   31.6671   38.0129   26.3634   26.8527

1282
  mean         55.8749   64.8499   16.5673   66.2608   30.9917    93.886   73.3865   76.2723   98.7819   94.6906
  std. dev.    33.4539   31.3046   26.5184   34.0328   40.0241   25.8854   34.0586   37.3914   24.1962   23.1281

792
  mean         35.7904    0.3919   28.2216   18.3246    0.3249    5.5114   55.5601         0    5.0054    6.0571
  std. dev.    38.7378    3.2694   37.5151   29.8946    2.2568   16.0232   39.3239   33.6482   15.3031   17.1121

793
  mean         42.2233    0.5907   30.5369   19.5143     1.888    6.6139   57.9452    0.0458    10.317    7.8067
  std. dev.    39.0336    5.7145   38.8996   31.6595    7.1436   18.7727   39.3598    0.8548   23.5768   21.6508

1311
  mean         56.0369    0.2849   41.4883   46.2657   24.6682   90.5723   66.7441   71.5693   93.3901   86.3634
  std. dev.    35.2303     2.116    37.372   37.9336   34.3156   22.7496   34.9617   39.4905   27.2115   25.0109

1312
  mean         55.8579         0   45.4085   44.0758   24.1156   91.2996   70.9198   69.4904   94.5154   87.2167
  std. dev.    35.6481   42.8724   37.1902   38.9545   35.1611   24.1879   34.4515   40.1005   25.3219   29.0209

1313
  mean         46.0167         0   46.7085   42.9864   23.1771   87.8571   65.5636   63.1402   88.2753    85.759
  std. dev.    37.1577   42.4937   37.8103   40.4259   34.8008   28.3789   35.2215   37.7268   25.6088   33.4246

1328
  mean         58.9132   59.5038   17.4409   61.7721   27.6931   94.2787    72.759   64.6466   99.6397   91.9056
  std. dev.    32.9788   36.4978   29.6005   38.1051   37.5612   29.3323   31.0113   37.1114   27.2778   32.0762

1329
  mean         54.7909    69.559   11.8342   66.4042   28.5875   95.1079   69.6152   72.6184  102.5713   93.9698
  std. dev.    33.5573   29.4772   21.1203   33.0137   37.6637   28.7335   32.1615   38.1251   26.9123   28.9145

1330
  mean         43.7712   65.0697   17.5475   64.9498   28.6376   93.8654   67.7523   75.0369  100.1324   93.3586
  std. dev.    35.6854   31.3278   28.2246    33.406   38.9763   26.6363   34.5476   38.5414   23.7854   25.1421

1846
  mean         49.5416    6.5258   23.5311    14.439    7.0832   20.5715   36.2287   70.0597   12.9548   13.3992
  std. dev.    37.2768   20.9879   35.7078   29.3926   21.7794   33.9384   42.9698   39.4498   28.8069    28.469

1360
  mean         54.6116         0   40.8729   41.2345   24.5687    89.743   72.6815   71.0207   94.9393   86.4322
  std. dev.     35.452   43.7194   37.4919   38.0676   34.5819   28.0532   37.1752   38.9249   26.8774   31.8113

1361
  mean         55.9396         0   42.9172   40.5378   21.9875   86.0569   72.0107   64.5309   87.3609   83.1226
  std. dev.    35.5553   43.0512   39.0985   40.0394   34.0611   31.1655   35.9313   38.7627   26.9384   35.8196

1367
  mean          2.7915   24.6739   38.1011   38.4303    20.595    84.857   30.6561    9.8951   93.9903   76.2355
  std. dev.    11.8459   32.5377    36.384   39.9135   34.1606   38.1074   36.3265   19.9806   29.1486   42.5797

1377
  mean         42.7778   62.9391   15.2929   62.8679   27.3422   94.3775   68.8015   75.7829  101.0638    88.303
  std. dev.    34.6787   31.0399   26.9762   33.4934   37.1766   29.1961   33.1875   37.4593   28.2032   32.1378

888
  mean         11.5732    5.1074   27.9082   36.6986   11.7969   38.8411   35.7054         0   58.9287   35.5569
  std. dev.    24.0719    16.259   36.3071   38.5617   25.3622   37.0392   37.6039   35.9649   41.9476   34.0627

1415
  mean          7.7679   16.8157   43.0749   43.3494   25.8577   84.5404   32.3125   19.5169   87.5367   81.9399
  std. dev.    21.0837   29.8929   38.0034   38.9125   35.1392   32.9333   36.4683   26.7881   29.6355   35.6284

1416
  mean          2.9783   23.6902   41.1143   44.0876    26.795   85.6217   26.0951    6.3162   90.9142   80.8026
  std. dev.    12.0649   30.9998   38.0013   39.3547   35.7884   34.3995   34.8302   15.7262   27.2602   37.5829

1417
  mean          1.5649   39.1347   36.1827   45.5563   25.9415   84.9146   30.1541     5.484   91.7356   78.3151
  std. dev.     8.1496   34.6737   37.1859    40.053   35.0246   36.2255    35.456   14.9146   27.1161   38.4508

936
  mean           4.955   14.1658   27.5206   44.8888   18.0096   59.3169   32.9013    0.0456   78.6155   57.6214
  std. dev.    16.7237   23.7436    35.843   38.9209   30.0099   36.2668   37.8845     0.853   36.1596   35.6842

1967
  mean         60.6156     58.92    63.888   34.9227   28.5896   68.0112    64.217   62.4527   63.5625    5.8129
  std. dev.    42.9401   42.0689    40.793   42.6442   41.9828   39.9633   40.7968   44.0723   41.9945   21.6001

1464
  mean          7.1031   18.6626   37.9325   43.5687    29.037   71.3124   27.6414   15.1007   70.5162   70.0582
  std. dev.    18.7326   29.8563   37.6102   38.5624   36.3509   34.3182   34.2993   24.4622   29.9773   34.6177

985
  mean          0.8688   13.9262   34.5354   43.0479   28.3806   80.5046   41.5022         0    96.986   78.2982
  std. dev.     4.9137   23.3224   37.1869   39.7453   37.9929   31.6917   39.9924   43.9083    26.706   31.1508

986
  mean          3.6143    5.9306   38.4377   40.3824   24.0662   81.1176   46.3551    0.0002   93.1976    80.527
  std. dev.    12.0764   15.9979   38.2579   38.8945   34.6991   30.5653   38.8178    0.0559    31.467   30.4585

label
  eight              1  908.2548   57.7837  105.2785  126.6252    5.1375  156.9285    1.9909     1.001         1
  one           3.0001    1.0098   43.6107  179.7937  111.3964  295.7977   72.7367    2.3391  206.9249  324.3908
  three              1         1   13.8294   30.6009         2    2.0351  826.9049         1         1    3.6297
  six                1         1         3    7.4259    1.0003    66.711   17.8662         1  420.9964    5.0001
  five         78.4467         2   53.8493   12.5549   46.2155    2.9182   218.552 1024.4626         1    1.0009
  two                1     3.999    9.8198   91.6619   89.0059   10.1036    3.7804         1   13.9249   68.7045
  four        925.0674    2.9758   95.9612   39.7179    31.212         1  258.4063   27.6594         1         1
  zero               2    2.1182   13.8636   67.4253   75.8042  263.9015   50.1144         1  217.9453  262.8275
  seven              1    3.0091   42.0877    3.9038  105.9993         1         1         1         1         1
  nine               1    7.2766  149.5486    4.7114   33.7237    1.0089         1         1    4.7308         1
  [total]    1014.5142  932.6433  483.3541  543.0743  622.9825  649.6134 1607.2893  1062.452  869.5232  669.5536


Time taken to build model (percentage split) : 13.85 seconds

Clustered Instances

 0       600 ( 14%)
 1       529 ( 12%)
 2       244 (  6%)
 3       228 (  5%)
 4       215 (  5%)
 5       305 (  7%)
 6       775 ( 18%)
 7       516 ( 12%)
 8       421 ( 10%)
 9       472 ( 11%)


Log likelihood: -385.49284

